src/Lights/directionalLight.ts

```diff
@@ -204,10 +210,19 @@ export class DirectionalLight extends ShadowLight {
      */
     public transferToEffect(effect: Effect, lightIndex: string): DirectionalLight {
         if (this.computeTransformedInformation()) {
-            this._uniformBuffer.updateFloat4("vLightData", this.transformedDirection.x, this.transformedDirection.y, this.transformedDirection.z, 1, lightIndex);
+            if (this.getScene().useRightHandedSystem) {
+                this._uniformBuffer.updateFloat4("vLightData", -this.transformedDirection.x, -this.transformedDirection.y, -this.transformedDirection.z, 1, lightIndex);

```

Comment:
Why should we negate light direction when using right handed system?

Reply:
Because the light computation is in view space https://github.com/BabylonJS/Babylon.js/blob/master/src/Shaders/ShadersInclude/lightFragment.fx#L14
and
https://github.com/BabylonJS/Babylon.js/blob/master/src/Shaders/default.fragment.fx
and switching to RH flips  the direction for the dots. 
More empirically, the lighting is wrong when not negating vLightData. I can clearly see why for x and z. It's more fuzzy for y.